llvm/lib/Target/AMDGPU/AsmParser/AMDGPUAsmParser.cpp

```diff
@@ -4588,28 +4588,40 @@ bool AMDGPUAsmParser::validateOpSel(const MCInst &Inst) {
 
   uint64_t TSFlags = MII.get(Opc).TSFlags;
 
-  if (isGFX940() && (TSFlags & SIInstrFlags::IsDOT)) {
-    int OpSelIdx = AMDGPU::getNamedOperandIdx(Opc, AMDGPU::OpName::op_sel);
-    if (OpSelIdx != -1) {
-      if (Inst.getOperand(OpSelIdx).getImm() != 0)
-        return false;
+  if (TSFlags & SIInstrFlags::IsDOT) {
+    // For all DOT instructions on GFX940, or VOP3P DOT instructions on all
+    // targets, i.e. v_dot2_*(except on pre-GFX940), v_dot4_* and v_dot8_*,
```

Comment:
I have checked gfx10 and DOT2 supports opsel, at least for 3 instructions below...

Reply:
Ok, I'll look into this.